src/aiida/storage/psql_dos/backend.py

```diff
@@ -477,3 +477,101 @@ def get_info(self, detailed: bool = False) -> dict:
+        # (locked means that possibly a maintenance operation is running that could interfere with the backup)
+        try:
+            ProfileAccessManager(self._profile).request_access()
+        except exceptions.LockedProfileError as exc:
+            raise exceptions.StorageBackupError('The profile is locked!') from exc
+
+        # step 1: first run the storage maintenance version that can safely be performed while aiida is running
+        self.maintain(full=False, compress=True)
+
+        # step 2: dump the PostgreSQL database into a temporary directory
+        pg_dump_exe = manager.exes.get('pg_dump', 'pg_dump')
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            psql_temp_loc = pathlib.Path(temp_dir_name) / 'db.psql'
+
+            env = os.environ.copy()
+            env['PGPASSWORD'] = cfg['database_password']
+            cmd = [
+                pg_dump_exe,
+                f'--host={cfg["database_hostname"]}',
+                f'--port={cfg["database_port"]}',
+                f'--dbname={cfg["database_name"]}',
+                f'--username={cfg["database_username"]}',
+                '--no-password',
+                '--format=p',
+                f'--file={psql_temp_loc!s}',
+            ]
+            try:
+                subprocess.run(cmd, check=True, env=env)
+            except subprocess.CalledProcessError as exc:
+                raise backup_utils.BackupError(f'pg_dump: {exc}')
+
+            if psql_temp_loc.is_file():
+                STORAGE_LOGGER.info(f'Dumped the PostgreSQL database to {psql_temp_loc!s}')
+            else:
+                raise backup_utils.BackupError(f"'{psql_temp_loc!s}' was not created.")
+
+            # step 3: transfer the PostgreSQL database file
+            manager.call_rsync(psql_temp_loc, path, link_dest=prev_backup, dest_trailing_slash=True)
+
+        # step 4: back up the disk-objectstore
+        backup_utils.backup_container(
+            manager, container, path / 'container', prev_backup=prev_backup / 'container' if prev_backup else None
+        )
+
+        # step 5: back up aiida config.json file (strip other profiles!)
+        try:
+            config = get_config()
+            profile = config.get_profile(self.profile.name)  # Get the profile being backed up
+            with tempfile.TemporaryDirectory() as tmpdir:
+                filepath_config = pathlib.Path(tmpdir) / 'config.json'
+                backup_config = Config(str(filepath_config), {})  # Create empty config at temporary file location
+                backup_config.add_profile(profile)  # Add the profile being backed up
+                backup_config.store()  # Write the contents to disk
+                manager.call_rsync(filepath_config, path)
+        except (exceptions.MissingConfigurationError, exceptions.ConfigurationError):
+            STORAGE_LOGGER.warning('aiida config.json not found!')
```

Comment:
Sorry that I am coming to this only now, but should this maybe be done as the first step? Since you are essentially ignoring the error, _if_ it were to occur, there would be a backup without config file and so it wouldn't be clear to which profile it belongs. Maybe it is better to put this as the first step and if it fails, simply stop the backup operation.

Reply:
Good point. moved to the beginning.